addons/mail/controllers/discuss/translation.py

```diff
@@ -0,0 +1,57 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import requests
+from odoo.http import request, route, Controller
+
+
+# TODO KFR: Refacto (abstraction) + Errors in response
+class DiscussTranslationController(Controller):
+    languages = dict()
+
+    @route("/discuss/translation/translate", type="json", auth="user")
+    def translate(self, body):
+        result = dict.fromkeys(["source", "value", "errors"])
+        if source_code := self._detect_source(body):
```

Comment:
`:=` is a py3.10 feature, and I'm pretty sure we must support py3.8, so better not use it yet

Reply:
This operator works with Python 3.8 and is already used in some standard Odoo modules such as account and base

Reply:
Ok, I thought it was 3.10 but it's indeed 3.8, so can ignore comment